kubejs/server_scripts/gregtech/circuits.js

```diff
@@ -72,6 +72,110 @@ ServerEvents.recipes(event => {
             .EUt(245760)
     }
 
+    for (const javaRecipe of event.findRecipes({ mod: "gtceu" })) {
+        /** @type {import("../../dx/typings/GTJSONRecipe.d.mts").GTJSONRecipe} */
+        let recipe = JSON.parse(javaRecipe.json.toString())
+
+        // Filter out non-GT-machine recipes
+        if(!recipe?.type.startsWith("gtceu:"))
+            continue
+
+        // Filter out non-item recipes
+        if (!(recipe.duration && recipe.inputs?.item && recipe.outputs))
+            continue
+
+        // Filter out recipes without advanced smd
+        if(!recipe.inputs.item.some(i =>
+            i.content.type === "gtceu:sized" &&
+            "item" in i.content.ingredient &&
+            i.content.ingredient.item.startsWith("gtceu:advanced_smd_")
+        )) continue
+
+        // Filter out recipes without eu
+        if(!recipe.tickInputs?.eu)
+            continue
+        let eut = recipe.tickInputs.eu[0].content
+        if(!eut)
+            continue
+
+        // Extract inputs and outputs data
+        let [newInputItems, newOutputItems] = [recipe.inputs.item, recipe.outputs?.item].map(items =>
+            items && items.map(i => {
+                let c = i.content
+                if (c.type !== "gtceu:sized" || "type" in c.ingredient)
+                    throw new Error("Cannot generate complex SMD recipes")
+                let ing = c.ingredient
+                return {
+                    id: "tag" in ing? "#"+ing.tag : ing.item,
+                    amount: c.count
+                }
+            })
+        )
+        let [newInputFluids, newOutputFluids] = [recipe.inputs.fluid, recipe.outputs?.fluid].map(items =>
```

Comment:
I *think* there is probably a much less verbose way to implement this, using `GTRecipe.copy`, and modifying the resulting recipe. I think the only bit that isn't straightforward might be adding the resulting recipe to the game.

This avoids needing to know how to replicate all the details of the existing recipes.